plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1091,6 +1141,13 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
 
       my->applied_transaction_connection = my->chain->applied_transaction.connect(
             [this]( std::tuple<const transaction_trace_ptr&, const signed_transaction&> t ) {
+               if (auto dm_logger = my->chain->get_deep_mind_logger()) {
+                  fc_dlog(*dm_logger, "APPLIED_TRANSACTION ${block} ${traces}",
+                     ("block", chain().head_block_num() + 1)
+                     ("traces", chain().to_variant_with_abi(std::get<0>(t), abi_serializer::create_yield_function(fc::microseconds(config::dmlog_abi_serializer_max_time_us))))
```

Comment:
So, we never really bothered with this problem since are motto as always been: "let it crash, and deal with it when it happens". In fact, in never happen yet that we got an exception being thrown here. Not saying I would not change this.

I would need to think, doing it without ABI might be a solution, I just fear it goes unnoticed  this way. At the same time when the ABI is wrong or inexistent, it already keep it in hex form. I'll think of that today.

As for the 5s, like mentioned in a comment below, it's really arbitrary, enough in our eye to let the ABI serializer finish its job no matter what. Do you have a suggestion?

Reply:
It will not crash in all cases. In a number of these places it will be silently ignored. If you want a crash when this happens then I would add a try-catch, log why, and stop/crash. In this particular location it will ignore the exception and will not publish to the channel nor will any other signal slots be signaled. So it will just silently be ignored.

Reply:
You say it has "never happened yet" I wonder if it might have and you have not noticed.

Reply:
Well, you are right that it’s possible that it went unnoticed already. However, that would mean we are missing some transactions, or some data in general, which I assume would have been reported. But I agree, it’s a possibility. I think in our reading code, we should assert that transaction traces match with what we have in the block. That could make a good safe guard.

Now, about the serializer throwing exceptions, I’ll go with a try catch and write the transaction without ABI serializer work as well as logging an error. I think it’s good compromise, will make life easier.

Now, are you ok if I create an helper for that, there is around 8 such call right now, probably in Controller? Or do you prefer that I avoid that and duplicate the code.

Once we agree here, I’ll make the necessary changes this afternoon as well as taking care of your other comments.

Reply:
We don't want the duplicate code. I hate that controller becomes the dumping ground for everything because it is convenient, however, the `to_variant_with_abi` is already there so it makes sense to add a `maybe_to_variant_with_abi` or something like that, which does the same thing with a try-catch that falls back to copying `obj` to `pretty_output` if exception. Note that even that copy could be expensive.

Reply:
Yeah, the controller is such a beast, that's why I fear adding new stuff to it :)

Did the changes with `maybe_to_variant_with_abi`. I think we can afford the expensive copy, assuming this should not be triggered often and that it costs less than doing the ABI serialiazation to variant anyway.

Will see with the new log if this is really happening or not it seems. Also update all occurrences as well as the comments.